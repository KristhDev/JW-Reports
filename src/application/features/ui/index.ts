export { default as uiReducer, UI_INITIAL_STATE, setIsDataExporting, setIsKeyboardVisible, setOldDatetimePicker } from './slice';
export * from './types';