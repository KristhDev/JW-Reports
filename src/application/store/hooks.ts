import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();