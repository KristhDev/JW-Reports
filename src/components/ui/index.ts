export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Title } from './Title';