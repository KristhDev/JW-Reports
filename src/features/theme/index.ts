export { default as themeReducer, setTheme } from './slice';