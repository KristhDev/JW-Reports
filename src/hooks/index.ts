export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';