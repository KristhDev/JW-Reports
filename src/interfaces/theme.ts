export type Theme = 'dark' | 'light' | 'default';

/* Defining the interface for the ThemeState. */
export interface ThemeState {
    colors: Colors;
    deviceTheme: Theme;
    isLoadedTheme: boolean;
    selectedTheme: Theme,
    theme: Theme,
}

/* Defining the interface for the Colors object. */
export interface Colors {
    background: string;
    bottom: string;
    button: string;
    buttonDark: string;
    card: string;
    contentHeader: string;
    focus: string;
    header: string;
    headerText: string;
    icon: string;
    inputText: string;
    linkText: string;
    modal: string;
    modalText: string;
    navbar: string;
    text: string;
    titleSecondary: string;
    titleText: string;
}

/**
 * SetThemePayload is an object with a property called theme that is of type Theme
 * @property {Theme} theme - The theme to set.
 */
export type SetThemePayload = {
    theme: Theme
}