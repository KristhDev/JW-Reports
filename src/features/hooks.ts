import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;