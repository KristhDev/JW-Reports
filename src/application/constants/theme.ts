import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];