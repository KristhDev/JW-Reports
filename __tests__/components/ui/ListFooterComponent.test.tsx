import React from 'react';
import { render, screen } from '@testing-library/react-native';

import { ListFooterComponent } from '../../../src/components/ui';

import { useTheme } from '../../../src/hooks';

import { darkColors } from '../../../src/theme';

jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <ListFooterComponent /> component', () => {
    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors }
    });

    beforeEach(() => {
        render(
            <ListFooterComponent
                showLoader={ true }
                marginTopPlus={ false }
            />
        );
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should render loader', () => {
        const loader = screen.getByTestId('loader');

        expect(loader).toBeTruthy();
    });

    it('should not render loader', () => {
        render(
            <ListFooterComponent
                showLoader={ false }
                marginTopPlus={ true }
            />
        );

        expect(screen.toJSON()).toBeNull();
    });
});