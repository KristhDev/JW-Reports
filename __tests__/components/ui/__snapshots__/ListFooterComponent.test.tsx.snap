// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ListFooterComponent /> component should to match snapshot 1`] = `
<ActivityIndicator
  color="#C0A7E1"
  size="large"
  style={
    {
      "marginTop": 133.4,
    }
  }
  testID="loader"
/>
`;
